zaps/contracts/OlympusZapManager.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: WTFPL
+pragma solidity ^0.8.0;
+
+import "./interfaces/IBondDepository.sol";
+
+import "./libraries/Ownable.sol";
+
+contract OlympusZapManager is Ownable {
+
+
+    /////////////// storage ///////////////
+
+    address public staking = 0xFd31c7d00Ca47653c6Ce64Af53c1571f9C36566a;
+
+    address public constant OHM = 0x383518188C0C6d7730D91b2c03a03C837814a899;
+
+    address public sOHM = 0x04F2694C8fcee23e8Fd0dfEA1d4f5Bb8c352111F;
+
+    address public wsOHM = 0xCa76543Cf381ebBB277bE79574059e32108e3E65;
+
+    // IE DAI => DAI bond depo
+    mapping(address => address) public principalToDepository;
+
+    ///////////// public logic ////////////
+
+    function deposit(
+        address _principal,
+        uint _amount,
+        uint _maxBondPrice
+    ) external returns ( uint ) {
+        IBondDepository depository = IBondDepository( principalToDepository[ _principal ] );
+        // make sure market exists for given principal/toToken
+        require( principalToDepository[ _principal ] != address(0), "bonding market doesn't exist");
+        // buy bond on the behalf of user
+        depository.deposit( _amount, _maxBondPrice, msg.sender );
```

Comment:
Err the only thing here, is that msg.sender is going to be the zap contract. the zap contract is gonna be mega rich with Ohm. lol. 

You need to pass the `_depositor` address down from the Zap Method